### PR TITLE
index: Add dropdown for picking release

### PR DESCRIPTION
**Pull request by FilipZajdel**
It allows the user to choose the extensions's release. Pushing the VSCode button will redirect to a url that has that particular release.
[Screencast from 22. mars 2024 kl. 15.30 +0100.webm](https://github.com/nrfconnect/ncs-app-index/assets/35106656/325adbd1-03c8-42c7-bb9d-06bef594db05)

### CODE REVIEW

**Comment by datenreisender** [comment]
Two more comments: 

1. When clicking on “Instructions”, the selected branch is not respected anymore, because it is still fixed there: https://github.com/nrfconnect/ncs-app-index/blob/391e85146ea9b75f3fd9b265fb611b321e5efb75/site/src/app/InstructionsDialog.tsx#L47 Is there a plan to also change that?

2. BTW, that line seems to be wrong. Even though it says `latestRelease` in the line, it seems to always use the oldest release. Could you maybe also check and fix that (potentially in a different PR)?

**Comment by bencefr** [comment]
The dropdown listing the releases shows "main" and "ncs-example-app v2.6.0"... Shouldn't that only be "v2.6.0"? The name before the tag is not just noise, but it doesn't fit and the important bit is not visible in the input component.

**Review by datenreisender — APPROVED** [review]
That the currently selected value is not in the select list (because you use `hideSelectedOptions={true}`) confused me a bit, but if you want to do it like that, it is fine by me.

BTW, in JSX you can also simplify `hideSelectedOptions={true}` to just be `hideSelectedOptions`, in the same way as boolean attributes work in HTML.

File: site/src/app/InstructionsDialog.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ function InstructionsDialog({ app, close }: Props): JSX.Element {
 
                     <CodeBlock
                         text={`west init -m "${app.repo}" --mr ${
-                            latestRelease?.tag ?? '<latest tag>'
+                            latestRelease ?? '<latest tag>'
```
Comment (datenreisender) [inline site/src/app/InstructionsDialog.tsx:47]: With the changed assignment above `latestRelease` will always be defined, so the `?? '<latest tag>'` could be removed here. But it is also ok if you want to keep it as an additional safety measure.

**Review by datenreisender — APPROVED** [review]
Good work!

**Pull request merged**